src/go/rpk/pkg/cli/redpanda/config.go

```diff
@@ -252,3 +280,39 @@ func getSelfIP() (net.IP, error) {
 		return nil, errors.New("multiple private v4 IPs found, please select one with --self")
 	}
 }
+
+func parseAdvertisedKafka(adv string, defHost string) (config.NamedSocketAddress, error) {
+	host, port := defHost, config.DefaultKafkaPort
+	if adv != "" {
+		_, hostport, err := vnet.ParseHostMaybeScheme(adv)
+		if err != nil {
+			return config.NamedSocketAddress{}, err
+		}
+		host, port = vnet.SplitHostPortDefault(hostport, config.DefaultKafkaPort)
+	}
+	if host == "0.0.0.0" {
+		return config.NamedSocketAddress{}, errors.New("unexpected kafka advertised address '0.0.0.0' this will cause the broker to fail during startup validation")
+	}
+	return config.NamedSocketAddress{
+		Address: host,
+		Port:    port,
+	}, nil
+}
+
+func parseAdvertisedRPC(adv string, defHost string) (*config.SocketAddress, error) {
```

Comment:
nit: Could probably merge this function with the previous if you add a port field and name field (name being a string for the errors)

but separate is fine too

Reply:
I thought of that, but Kafka config receives a `NamedSocketAddress` while RPC receives a `SocketAddress`, so I had to do a type conversion so I ended up splitting the function

Reply:
ah, i c